tests/unit/generator/lib.php

```diff
@@ -0,0 +1,162 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+global $DB;
+
+require_once $CFG->dirroot.'/mod/turnitintooltwo/turnitintooltwo_assignment.class.php';
+require_once $CFG->dirroot.'/mod/turnitintooltwo/turnitintooltwo_user.class.php';
+include_once $CFG->dirroot.'/course/lib.php';
+require_once $CFG->dirroot.'/webservice/tests/helpers.php';
+/**
+* Turnitintooltwo module data generator class
+*
+* @category  Test
+* @package  mod_turnitintooltwo
+* @copyright  2017 David Hatton
+* @license  http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+*/
+abstract class test_lib extends advanced_testcase
+{
+    /**
+     * Create a test part on the specified assignment.
+     *
+     * @param string $modname Module name (turnitintool or turnitintooltwo)
+     * @param int $assignmentid Assignment Module ID
+     * @param int $number_of_parts - The number of parts to create
+     *
+     * @return array $parts_created - list of part ids that have been added to the assignment
+     */
+    public static function make_test_parts($modname, $assignmentid, $number_of_parts)
```

Comment:
Actually, the more I think about it, the more I think `static` isn't required. A normal test case would be extending `advanced_testcase`, like so:

```
class test extends advanced_testcase {
    ....
}
```

But with this abstract class the idea looks to be to extend this instead, which in turn extends `advanced_testcase`. So tests could use methods defined in this abstract class like:

```
class test extends advanced_testcase {
    public function test_one () {
        $this->make_test_parts('foo', 123, 1);
    }
}
```

The static in my opinion won't add anything to the code other than saving a few charcters.